firebase-admin/runtime/src/main/java/io/quarkiverse/googlecloudservices/firebase/admin/runtime/FirebaseSessionCookieManager.java

```diff
@@ -0,0 +1,192 @@
+package io.quarkiverse.googlecloudservices.firebase.admin.runtime;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jboss.logging.Logger;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseAuthException;
+import com.google.firebase.auth.FirebaseToken;
+import com.google.firebase.auth.SessionCookieOptions;
+
+import io.quarkus.security.credential.TokenCredential;
+import io.quarkus.security.identity.IdentityProviderManager;
+import io.quarkus.vertx.http.runtime.security.QuarkusHttpUser;
+import io.vertx.core.http.Cookie;
+import io.vertx.core.http.CookieSameSite;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Manages using cookie based authentication (see <a href="https://firebase.google.com/docs/auth/admin/manage-cookies#java_2">
+ */
+@ApplicationScoped
+public class FirebaseSessionCookieManager {
+
+    private static final Logger LOG = Logger.getLogger(FirebaseSessionCookieManager.class.getName());
+
+    @ConfigProperty(name = "quarkus.google.cloud.firebase.auth.session-cookie.name")
+    String sessionCookieName;
+
+    @ConfigProperty(name = "quarkus.google.cloud.firebase.auth.session-cookie.check-revoked")
+    boolean checkRevoked;
+
+    @ConfigProperty(name = "quarkus.google.cloud.firebase.auth.session-cookie.expiration-duration")
+    String expirationDuration;
+
+    @ConfigProperty(name = "quarkus.google.cloud.firebase.auth.session-cookie.validate-token")
+    boolean validateToken;
+
+    @ConfigProperty(name = "quarkus.google.cloud.firebase.auth.session-cookie.minimum-token-validity")
+    Optional<String> minimumTokenValidity;
+
+    @ConfigProperty(name = "quarkus.google.cloud.firebase.auth.session-cookie.login-api-path")
+    Optional<String> loginPath;
+
+    @ConfigProperty(name = "quarkus.google.cloud.firebase.auth.session-cookie.logout-api-path")
+    Optional<String> logoutPath;
```

Comment:
You can inject the configguration object itself, would be more easier instead of injecting each property individually.

Reply:
Fixed